src/App.tsx

```diff
@@ -1,24 +1,299 @@
+import classNames from 'classnames';
+
 import { UserWarning } from './UserWarning';
+import { FilterStatus, Todo } from './types/Todo';
+import {
+  addTodo,
+  deleteTodo,
+  getTodos,
+  patchTodoStatus,
+  patchTodoTitle,
+} from './utils/todosMethods';
+import { Error } from './components/Error';
+import { TodoForm } from './components/TodoForm';
+import { TodoList } from './components/TodoList';
+import { TodoFilter } from './components/TodoFilter';
+
+const USER_ID = 11220;
+
+function getPreparedTodo(todosFilter: Todo[], filterField: FilterStatus) {
+  return todosFilter.filter(todo => {
+    switch (filterField) {
+      case FilterStatus.Completed:
+        return todo.completed;
 
-const USER_ID = 0;
+      case FilterStatus.Active:
+        return !todo.completed;
+
+      default:
+        return todo;
+    }
+  });
+}
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [todoTitle, setTodoTitle] = useState('');
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [isLoading, setIsLoading] = useState(true);
+  const [errorMessage, setErrorMessage] = useState('');
+  const [filter, setFilter] = useState(FilterStatus.All);
+  const [newTodoTitle, setNewTodoTitle] = useState('');
+
+  const clearError = () => {
+    setErrorMessage('');
+  };
+
+  const applyFilter = (filterField: FilterStatus) => {
+    setFilter(filterField);
+  };
+
+  const preparedTodo = useMemo(() => {
+    return getPreparedTodo(todos, filter);
+  }, [todos, filter]);
+
+  const createTodo = () => {
+    setTempTodo({
+      id: 0,
+      userId: USER_ID,
+      title: todoTitle,
+      completed: false,
+    });
+
+    return addTodo({
+      title: todoTitle,
+      completed: false,
+      userId: USER_ID,
+    })
+      .then(newTodo => {
+        setTodos(previousTodo => [...previousTodo, newTodo]);
+      })
+      .catch(err => {
+        setErrorMessage('Can`t add new todo');
+        throw err;
+      })
+      .finally(() => setTempTodo(null));
+  };
+
+  const removeTodo = (todoId: number) => {
+    setIsLoading(true);
+
+    return deleteTodo(todoId)
+      .then(() => {
+        setTodos(previousTodos => {
+          return previousTodos.filter(todo => todo.id !== todoId);
+        });
+      })
+      .catch((err) => {
+        setTodos(todos);
+        setErrorMessage('Can`t delete todo');
+        throw err;
+      })
+      .finally(() => setIsLoading(false));
+  };
+
+  const clearCompleted = () => {
+    setIsLoading(true);
+    setTodos(prevTodos => {
+      return prevTodos.filter(todo => {
+        if (todo.completed) {
+          deleteTodo(todo.id)
+            .catch((error) => {
+              setTodos(todos);
+              setErrorMessage('Unable to delete a todo');
+              throw error;
+            })
+            .finally(() => setIsLoading(false));
+
+          return false;
+        }
+
+        return true;
+      });
```

Comment:
it's bad case to change state and do some side actions side the callback
```suggestion
    todos.forEach(...);
```